config/example_test.go

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package xconfig_test
+
+import (
+	"log"
+
+	"github.com/m3db/m3x/config"
+)
+
+type config struct {
+	ListenAddress string `yaml:"listenAddress" validate:"nonzero"`
+}
+
+func ExampleLoadFile() {
+	cfg := &config{}
+	if err := xconfig.LoadFile(cfg, "/path/to/config/file"); err != nil {
+		log.Fatal(err)
```

Comment:
Examples without `// Output: ` are compiled iirc. Could you confirm this is being run in the test suite?

Reply:
If you include `// Output:` the test is both compiled an run as part of the test suite, if you don't include it the test is only compiled. I think only compiling it makes sense in this case because otherwise we would have to write a temporary config file to ensure the call to `config.LoadFile` works properly. I thought about it in this case but I wasn't aware of a way that I could the necessary setup code but have it hidden in the example, and I was concerned that including it in the example would make it more cumbersome and less clear. Let me know what you think though, I'm not certainly not fully convinced that this is the right approach.

Reply:
I tried writing up an example for this (below), and it looks like a unit test to me. IMO your approach is a good middle ground. Plus, go's stdlib seems to follow the same ~ e.g. https://golang.org/pkg/io/ioutil/#example_ReadFile


```
func ExampleLoadFile() {
  exampleData := byte[](`listenAddress: "0.0.0.0:12321"`)
  if err := ioutil.WriteFile("testFile", exampleData, ...) ; err != nil {
    log.Fatalf("unable to write example file: %v", err)
  }
  var cfg config
  if err := xconfig.LoadFile(&cfg, "exampleFile"); err != nil {
     log.Fatalf("unable to load file: %v", err)
  }
  fmt.Println("listenAddress: %s", cfg.ListenAddress)
  // Output: ...
}
```

Reply:
Ahh, nice. Seems the `ReadFile` example is a perfect model of what we want to do here. They include a `testdata` directory which contains the file itself. I think I'll add one here as well so we can run this as a unit test, but still hide the initial setup of actually writing the test file.

Reply:
sounds good!